File: backend/src/main/java/com/app/backend/dao/entities/Question.java
package com.app.backend.dao.entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;


import java.util.Date;

@Document
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Getter
@Setter
@Builder
public class Question {
    @Id
    private String id;
    private Date date;
    private boolean urgent;
    private String texte;
    @DBRef
    private User patient;



}

File: frontend/src/app/shared/Question/question.ts
import { RegisterUser } from "../register-user";
import { User } from "../user";


export class Question {
  id: string;
  date: Date;
  urgent: boolean;
  texte: string;
  patient: RegisterUser;

  constructor(id: string, date: Date, urgent: boolean, texte: string, patient: RegisterUser) {
    this.id = id;
    this.date = date;
    this.urgent = urgent;
    this.texte = texte;
    this.patient = patient;
  }
}
